File: frontend/lib/types/pitch.ts
export type PitchMedia = {
	media_id: number;
	pitch_id: number;
	url: string;
	media_type: string;
	order_in_description: number;
};

// return from backend
export type Pitch = {
	pitch_id: number;
	title: string;
	elevator_pitch: string;
	detailed_pitch: string;
	target_amount: number;
	raised_amount: number;
	profit_share_percent: number;
	status: string;
	investment_start_date: Date;
	investment_end_date: Date;
	created_at: Date;
	updated_at: Date;
	investment_tiers?: InvestmentTier[];
	media?: PitchMedia[];
};

export type InvestmentTier = {
	tier_id: number;
	pitch_id: number;
	name: string;
	min_amount: number;
	max_amount: number;
	multiplier: number;
	created_at: Date;
};

// for creating new pitch
export type NewPitch = {
	title: string;
	elevator_pitch: string;
	detailed_pitch: string;
	target_amount: number;
	profit_share_percent: number;
	investment_start_date: string;
	investment_end_date: string;
	investment_tiers: {
		name: string;
		min_amount: number;
		max_amount?: number | null;
		multiplier: number;
	}[];
};

// ✅ For updating, you might allow partial updates
export type UpdatePitch = Partial<NewPitch>;

File: backend/internal/model/database/pitch.go
package database

type Pitch struct {
	PitchID             *int64  `json:"id,omitempty"`
	CreatedAt           string  `json:"created_at"`
	Title               string  `json:"title"`
	ElevatorPitch       string  `json:"elevator_pitch"`
	DetailedPitch       string  `json:"detailed_pitch"`
	TargetAmount        uint64  `json:"target_amount"`
	ProfitSharePercent  float64 `json:"profit_share_percent"`
	UserID              string  `json:"user_id"`
	RaisedAmount        int64   `json:"raised_amount"`
	InvestmentStartDate string  `json:"investment_start_date"`
	InvestmentEndDate   string  `json:"investment_end_date"`
	UpdatedAt           *string `json:"updated_at,omitempty"`
	Status              string  `json:"status"`
}
